README.md

```diff
@@ -1,133 +1,616 @@
+> 
+> **NOTE: For Node.js only, currently versions >= 14 are supported, and are subject to change.**
 
-**WARNING: All interfaces are subject to change.**
+`smithy-typescript` includes the reference implementations of the [Smithy](https://smithy.io/) code generators for [TypeScript](https://www.typescriptlang.org/).
 
-## Generating a client
-
-This repository builds TypeScript declarations and JavaScript clients from Smithy
-models.
+For Client SDK code generation, the `typescript-client-codegen` plugin provides a framework for generating customizable TypeScript clients that can support multiple JavaScript platforms, including Node.js, Browser, and React-Native. See [the section on generating a client to see how to get started](#generating-a-client), or [the `typescript-client-codegen` documentation](#client-sdk-code-generation-typescript-client-codegen-plugin).
 
-The `smithy-typescript-codegen-test` package in this repos is an example of
-how to build a TypeScript client. The steps needed to build a TypeScript client
-are as follows:
+For Server SDK code generation, the `typescript-server-codegen` plugin provides a framework for generating server scaffolding at a higher level of abstraction and with type safety. More documentation can be found at in [the `typescript-server-codegen` documentation](#server-sdk-code-generation-typescript-server-codegen-plugin), or [smithy.io](https://smithy.io/2.0/ts-ssdk/index.html).
 
-1. Create a new directory for your package. For example, "foo-client".
-2. Create a `build.gradle.kts` file with the following contents:
+## Generating a client
 
-   ```kotlin
-    buildscript {
-        repositories {
-            mavenCentral()
+The Smithy TypeScript `typescript-client-codegen` code generator in this repository generates TypeScript clients from Smithy models, and can be built with both the idiomatic [Smithy CLI](#using-smithy-typescript-with-the-smithy-cli) or through [Gradle](#using-smithy-typescript-with-gradle).
+
+> The Smithy CLI is a prerequisite for this section when using the `smithy init` commands. See [the installation guide](https://smithy.io/2.0/guides/smithy-cli/cli_installation.html) for how to install the Smithy CLI. If installing the Smithy CLI is not preferred, the templates used can be found in the [Smithy Examples repository](https://github.com/smithy-lang/smithy-examples).
+
+For additional configuration, see [the `typescript-client-codegen` documentation](#client-sdk-code-generation-typescript-client-codegen-plugin) and [the documentation for `smithy-build.json`](https://smithy.io/2.0/guides/building-models/build-config.html).
+
+### Using Smithy TypeScript with the Smithy CLI
+
+Using the Smithy CLI, a new Smithy CLI project can be created using the default `smithy init` template. In this example, the project will be called `smithy-typescript-example-client`.
+
+```shell
+smithy init -o smithy-typescript-example-client
```

Comment:
I think it would be good to publish a new, fully-fleshed out weather client example to https://github.com/smithy-lang/smithy-examples and then walk through it. Then we can make sure the example client is tested and customers can get started with minimal modifications.

Reply:
I think so too (and have examples ready on my local machine). We can create a follow-up PR once the example is merged into smithy-examples.